### DISCUSSION TITLE
linkerd + helm test

### DISCUSSION THREAD

**Discussion by oocx** [linkerd/linkerd2#5696] [Q&A]
When I use "helm test" to run tests after deploying my app, the test container successfully runs and then terminates, but the sidecar that was injected into the test pod is still running, so "helm test" eventually times out.

Ideally, the sidecar should also terminate once the test container has terminated. 
If that's not possible, it should be possible somehow to prevent linkerd from injecting the sidecar into the test pod (without explicitly modifying my test pod, as I'd like to keep linkerd usage transparent for my dev teams).

Is this possible somehow?

**Comment by alpeb** [comment]
You could wrap the command in your test pod around `linkerd-await --shutdown`, which will send a termination signal to the proxy when the main command finishes. Take a look at the example in [linkerd-await](https://github.com/olix0r/linkerd-await) for how to set that up in a Dockerfile.

**Reply by oocx** [reply]
I could do that, but then I can simply add an annotation to the pod so that it does not get injected. My point was that I'd like linkerd usage to be transparent for my dev teams, they should not have to change anything in their pods to make them work with linkerd. 

I'd like to be able to simply install linkerd in my environment without breaking anything. With the current behavior, I have to modify all pods that are used for helm tests, so I need to make changes across several teams and repositories.

Thanks anyway, I did not know about linkerd-await and it looks like it might be useful for something else in the future.
